app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,539 @@
+
+      // Initiate discovery
+      if(!isWifiP2pEnabled()) {
+        requestEnableWifiP2pServices();
+        return true;
+      }
+
+      if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && !isLocationServicesEnabled()) {
+        requestEnableLocationServices();
+        return true;
+      }
+
+      final DeviceListFragment deviceListFragment = (DeviceListFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_device_list);
+      deviceListFragment.onInitiateDiscovery();
+      deviceListFragment.setSharedPreferenceUtil(sharedPreferenceUtil);
+      manager.discoverPeers(channel, new WifiP2pManager.ActionListener() {
+        @Override
+        public void onSuccess() {
+          showToast(LocalFileTransferActivity.this, R.string.discovery_initiated, Toast.LENGTH_SHORT);
+        }
+
+        @Override
+        public void onFailure(int reason) {
+          showToast(LocalFileTransferActivity.this, LocalFileTransferActivity.this.getString(R.string.discovery_failed) + " " + getErrorMessage(reason), Toast.LENGTH_SHORT);
+        }
+      });
+      return true;
+
+    } else if(item.getItemId() == R.id.menu_item_cancel_search) {
+      if(manager != null) {
+        cancelSearch();
+      }
+      return true;
+
+    } else {
+      return super.onOptionsItemSelected(item);
+    }
+  }
+
+
+  /* Helper methods used in the activity */
+  public void setDeviceAsFileSender() {
+    fileSendingDevice = true;
+  }
+
+  public boolean isFileSender() {
+    return fileSendingDevice;
+  }
+
+  public @NonNull ArrayList<Uri> getFileUriArrayList() {
+    return fileUriArrayList;
+  }
+
+  public void setWifiP2pEnabled(boolean wifiP2pEnabled) {
+    this.wifiP2pEnabled = wifiP2pEnabled;
+  }
+
+  public boolean isWifiP2pEnabled() {
+    return wifiP2pEnabled;
+  }
+
+  private String getErrorMessage(int reason) {
+    switch (reason) {
+      case WifiP2pManager.ERROR:           return getString(R.string.wifi_p2p_internal_error);
+      case WifiP2pManager.BUSY:            return getString(R.string.wifi_p2p_framework_busy);
+      case WifiP2pManager.P2P_UNSUPPORTED: return getString(R.string.wifi_p2p_unsupported);
+
+      default: return (getString(R.string.wifi_p2p_unknown_error) + " " + reason);
+    }
+  }
+
+  public void resetPeers() {
+    DeviceListFragment deviceListFragment = (DeviceListFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_device_list);
+    if(deviceListFragment != null) {
+      deviceListFragment.clearPeers();
+    }
+  }
+
+  public void resetData() {
+    DeviceListFragment deviceListFragment = (DeviceListFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_device_list);
+    if(deviceListFragment != null) {
+      deviceListFragment.clearPeers();
+    }
+  }
+  
+  static void showToast(Context context, int stringResource, int duration) {
+    Toast.makeText(context, context.getString(stringResource), duration).show();
```

Comment:
make this invoke the other showToast method

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/c6de4045c87b493399f574679e8b26614ed15132#diff-e1d59ae6e15dcb502e125fb6e243d15dR228)